src/components/internals/calendar-body/calendar-body.js

```diff
@@ -30,96 +30,114 @@ ClearSection.propTypes = {
   onClear: PropTypes.func,
 };
 
-export default class CalendarBody extends React.PureComponent {
-  static displayName = 'CalendarBody';
+export const CalendarBody = props => {
+  const [isFocused, setIsFocused] = React.useState(false);
 
-  static propTypes = {
-    inputRef: PropTypes.object.isRequired,
-    icon: PropTypes.string,
-    id: PropTypes.string,
-    inputProps: PropTypes.shape({
-      onBlur: PropTypes.func,
-      onFocus: PropTypes.func,
-    }),
-    isClearable: PropTypes.bool,
-    toggleButtonProps: PropTypes.shape({
-      onBlur: PropTypes.func,
-      onFocus: PropTypes.func,
-    }),
-    value: PropTypes.string,
-    isDisabled: PropTypes.bool,
-    isOpen: PropTypes.bool,
-    hasSelection: PropTypes.bool,
-    hasWarning: PropTypes.bool,
-    hasError: PropTypes.bool,
-    onClearPicker: PropTypes.func,
-    onClear: PropTypes.func,
-    placeholder: PropTypes.string,
-    horizontalConstraint: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl', 'scale']),
-  };
+  const { onFocus: inputOnFocus } = props.inputProps;
```

Comment:
Nit: onInputFocus. I prefer always prefixing with `on` or `handle`.

Reply:
addressed in my last commit